src/state/farms/fetchFarms.ts

```diff
@@ -8,88 +8,105 @@ import { FarmConfig } from 'config/constants/types'
-      const tokenAmountMc = tokenAmountTotal.times(lpTokenRatio)
-      const quoteTokenAmountMc = quoteTokenAmountTotal.times(lpTokenRatio)
+        // Amount of token in the LP that are staked in the MC (i.e amount of token * lp ratio)
+        const tokenAmountMc = tokenAmountTotal.times(lpTokenRatio)
+        const quoteTokenAmountMc = quoteTokenAmountTotal.times(lpTokenRatio)
 
-      // Total staked in LP, in quote token value
-      const lpTotalInQuoteToken = quoteTokenAmountMc.times(new BigNumber(2))
+        // Total staked in LP, in quote token value
+        const lpTotalInQuoteToken = quoteTokenAmountMc.times(new BigNumber(2))
 
-      const [info, totalAllocPoint] = await multicall(masterchefABI, [
-        {
-          address: getMasterChefAddress(),
-          name: 'poolInfo',
-          params: [farmConfig.pid],
-        },
-        {
-          address: getMasterChefAddress(),
-          name: 'totalAllocPoint',
-        },
-      ])
+        const [info, totalAllocPoint] = await multicall(masterchefABI, [
+          {
+            address: getMasterChefAddress(),
+            name: 'poolInfo',
+            params: [farmConfig.pid],
+          },
+          {
+            address: getMasterChefAddress(),
+            name: 'totalAllocPoint',
+          },
+        ])
 
-      const allocPoint = new BigNumber(info.allocPoint._hex)
-      const poolWeight = allocPoint.div(new BigNumber(totalAllocPoint))
+        const allocPoint = new BigNumber(info.allocPoint._hex)
+        const poolWeight = allocPoint.div(new BigNumber(totalAllocPoint))
 
-      return {
-        ...farmConfig,
-        tokenAmountMc: tokenAmountMc.toJSON(),
-        quoteTokenAmountMc: quoteTokenAmountMc.toJSON(),
-        tokenAmountTotal: tokenAmountTotal.toJSON(),
-        quoteTokenAmountTotal: quoteTokenAmountTotal.toJSON(),
-        lpTotalSupply: new BigNumber(lpTotalSupply).toJSON(),
-        lpTotalInQuoteToken: lpTotalInQuoteToken.toJSON(),
-        tokenPriceVsQuote: quoteTokenAmountTotal.div(tokenAmountTotal).toJSON(),
-        poolWeight: poolWeight.toJSON(),
-        multiplier: `${allocPoint.div(100).toString()}X`,
+        return {
+          ...farmConfig,
+          tokenAmountMc: tokenAmountMc.toJSON(),
+          quoteTokenAmountMc: quoteTokenAmountMc.toJSON(),
+          tokenAmountTotal: tokenAmountTotal.toJSON(),
+          quoteTokenAmountTotal: quoteTokenAmountTotal.toJSON(),
+          lpTotalSupply: new BigNumber(lpTotalSupply).toJSON(),
+          lpTotalInQuoteToken: lpTotalInQuoteToken.toJSON(),
+          tokenPriceVsQuote: quoteTokenAmountTotal.div(tokenAmountTotal).toJSON(),
+          poolWeight: poolWeight.toJSON(),
+          multiplier: `${allocPoint.div(100).toString()}X`,
+        }
       }
+
+      const timeout = new Promise((resolve) => {
+        const id = setTimeout(() => {
+          clearTimeout(id)
+          resolve(farmConfig)
+        }, 5000)
+      })
+
+      return Promise.race([timeout, farmFetch()])
```

Comment:
I wonder if that would be better handled by `Promise.allSettled` with proper rejection handling 🤔 
With this approach we create bunch of timers, not sure how much it affects performance, but seems a bit unnecessary

Reply:
Promise.allSettled needs promise to resolve or reject right ? In our issue it gets stuck that we get neither rejection nor resolve

I don't see any performance impact at first glance but you are right it needs to be further analyzed.

Reply:
aaah, in that case.... yeah, then race is better 👍